custom_components/tahoma/tahoma_device.py

```diff
@@ -141,35 +147,33 @@ def has_command(self, *commands):
 
     def select_state(self, *states):
         """Select first existing active state in a list of states."""
-        return next(
-            (
-                v
-                for k, v in self.tahoma_device.active_states.items()
-                if k in list(states)
-            ),
-            None,
-        )
+        if self.device.states:
+            return next(
+                (
+                    state.value
+                    for state in self.device.states
+                    if state.name in list(states)
+                ),
+                None,
+            )
+        return None
 
     def has_state(self, *states):
         """Return True if a state exists in self."""
         return self.select_state(*states)
 
     @Throttle(timedelta(seconds=1))
-    def should_wait(self):
+    async def should_wait(self):
         """Wait for actions to finish."""
-        exec_queue = self.controller.get_current_executions()
-        self._exec_queue = [e for e in self._exec_queue if e in exec_queue]
+        exec_queue = await self.client.get_current_executions()
+        self._exec_queue = [e.id for e in self._exec_queue if e in exec_queue]
```

Comment:
I am not 100% sure if this is functional. This should be tested more thoroughly.

Reply:
I hate this :D But @vlebourl like it :)

Reply:
@tetienne I can not read this myself as well... Thus not sure if it even works. It is still the old code.